onnxruntime/core/mlas/lib/compute.cpp

```diff
@@ -850,8 +850,21 @@ Return Value:
     const float* Input = WorkBlock->Input + n * D;
     float* Output = WorkBlock->Output + n * D;
 
+    constexpr size_t CacheLineSize = 64;
+    constexpr size_t ElementsPerCacheLine = CacheLineSize / sizeof(float);
```

Comment:
should these also go inside the ifdef? otherwise they may be unused.

consider [std::hardware_constructive_interference_size](https://stackoverflow.com/questions/39680206/understanding-stdhardware-destructive-interference-size-and-stdhardware-cons) as a portable way to get cache line size.

Reply:
`#ifdef` and `std::hardware_constructive_interference_size` were utilized.

Reply:
Revert to constant given the compiler used by CoreML EP doesn't identify the constexpr.